File: m4d/ClientApp/src/pages/playlist/PlayListPageModel.ts
import { jsonArrayMember, jsonMember, jsonObject } from "typedjson";

@jsonObject
export class PlayListSummary {
  @jsonMember(String) public id!: string;
  @jsonMember(String) public user!: string;
  @jsonMember(Number) public type!: number;
  @jsonMember(String) public name?: string;
  @jsonMember(String) public description?: string;
  @jsonMember(String) public data1?: string;
  @jsonMember(String) public data2?: string;
  @jsonMember(String) public created!: string;
  @jsonMember(String) public updated?: string;
  @jsonMember(Boolean) public deleted!: boolean;
}

@jsonObject
export class PlayListPageModel {
  @jsonArrayMember(PlayListSummary) public playLists!: PlayListSummary[];
  @jsonMember(Number) public type!: number;
  @jsonMember(String) public filteredUser?: string;
  @jsonMember(String) public data1Name!: string;
  @jsonMember(String) public data2Name!: string;
}

File: m4d/ViewModels/PlayListPageModel.cs
namespace m4d.ViewModels;

/// <summary>
/// Safe-to-serialise projection of a PlayList entry for the admin playlist index page.
/// Data1/Data2 are truncated to 50 chars for display.
/// </summary>
public class PlayListSummary
{
    public string Id { get; set; }
    public string User { get; set; }
    public int Type { get; set; }
    public string Name { get; set; }
    public string Description { get; set; }
    public string Data1 { get; set; }
    public string Data2 { get; set; }
    public DateTime Created { get; set; }
    public DateTime? Updated { get; set; }
    public bool Deleted { get; set; }
}

/// <summary>
/// Top-level model for the admin playlist Vue page.
/// Includes both active and deleted playlists; client toggles visibility.
/// </summary>
public class PlayListPageModel
{
    public List<PlayListSummary> PlayLists { get; set; }
    public int Type { get; set; }
    public string FilteredUser { get; set; }
    public string Data1Name { get; set; }
    public string Data2Name { get; set; }
}
